src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs

```diff
@@ -22,13 +23,29 @@ public class InMemoryExporter<T> : BaseExporter<T>
         where T : class
     {
         private readonly ICollection<T> exportedItems;
+        private readonly ExportDelegate onExport;
 
         public InMemoryExporter(ICollection<T> exportedItems)
         {
+            if (typeof(T) == typeof(Metrics.Metric))
+            {
+                throw new NotSupportedException("Exported Metrics are not trustworthy because they can continue to be updated after export. Recommend use ExportableMetricCopy.");
```

Comment:
I am not sure I follow the reason behind this throw...

Reply:
It is by design to force the use of the Copy. 
The thought behind this was to prevent users from creating the `InMemoryExporter<Metric>` because the `Metric` is known to be faulty. https://github.com/open-telemetry/opentelemetry-dotnet/pull/3198#discussion_r851547483

I've documented some scenarios where the `Metric` can and cannot be trusted: https://github.com/open-telemetry/opentelemetry-dotnet/issues/2361#issuecomment-1049175566

That being said, Prometheus and OTLP have some units tests where they need the actual `Metric` to provide to their own internals. (documented in the Summary above).

I'm eager to hear your perspective on this!
I'm trying to find the happy medium to support all these use cases while protecting users.
If we decide to allow users to create `InMemoryExporter<Metric>` we also need to inform/educate users on the risks.

Reply:
We don't need to prevent users from using the <Metric> one. A lot of tests are already using it, and working fine. (they understand the documented limitation of Metric that it may be updated after Export() call, but they are orchestrating things so as to not be affected)

This can be additional option, for InMemory users, who wants to test metrics, without being affected by the limitation that the memory could be updated.

Reply:
To unblock progress - lets split PR into sub Prs.
1. Add the additional option in InMemoryExporter to use MetricCopy, leave existing functionality untouched. i.e don't change unit tests, except the one breaking due to the metric-being-updated-after-export issue.
2. We can use the SIG meeting to see if there are strong reasons favoring/against keeping the current <Metric> exporter, and based on that keep it or remove it.